mitodl/bootcamp-ecommerce
# Personal prices for klasses

#### What are the relevant tickets?
Closes #139

#### What's this PR do?
Adds a `PersonalPrice` model to track user-specific pricing for a klass, and modifies klass price lookups to query this model first before getting the default price.

This PR assumes that the `amount_to_pay` field in a `WebhookRequest` is the personal user price for a klass. 

#### How should this be manually tested?
- Create a `Bootcamp`, `Klass` (`klass_key`=72713), and `Installment`.
- Send this `WebhookRequest` request, with `amount_to_pay` blank:
  ```curl -H 'Authorization: Basic faketoken' -H 'Content-type: text/plain' -d '{"date_of_birth":"","user_email":"staff@example.com","amount_to_pay":"","user_id":95292895,"submission_id":4621059,"award_id":72713,"award_cost":"1000","award_name":"TESTCAMP"}' http://192.168.99.101:8099/api/v0/fluidreview_webhook/```
- Log in via local EdX as `staff@example.com`
- Go to the payment page for the klass.  You should owe the same amount as you specified for the klass installment.
- Send the WebHookRequest again with `amount_to_pay` set to `'999.99'`.  Then reload the page. It should now say you owe $999.99 for the class.
- Send the WebHookRequest again with a different amount.  Then reload the page, you should see the new amount owed.
- Send the WebHookRequest again with a blank amount.  Then reload the page, you should see the default klass installment amount again.
